### PR TITLE
Basic Form App Project

### PR DESCRIPTION
**Pull request by Paul-Clue**
# What We Did:
- We used rails to create a form application using three different methods.
- First, we used the normal HTML form builder element.
- For our second method, we used the rails #form_tag method.
- And last, we used the rails #form_with method.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+  end
+
+  # def show
+  # end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (blackpintz) [inline app/controllers/users_controller.rb:10]: Remove this commented code.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (blackpintz) [inline app/controllers/users_controller.rb:3]: You can initialize user with just `@user = User.new`

**Review by blackpintz — CHANGES_REQUESTED** [review]
# Status: Changes Required 🛑

## Hello team 👋

**Great job on this project.**

### Highlights

- **Linters are properly configured.**
- **Professional README.**
- **Meets all project-specific requirements.**

### Mandatory Requirements

1). **Please remove the `if` statement logic in `views/users/new.html.erb`. You should not have logic (if statements, each loop) in your view files. View-related logic can be encapsulated in View helpers (`app/helpers/*_helper.rb`). See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from pathwright.**

2). **Check more in the comments below.**

### Optional

- **You can mention in your README file what route one should use for editing a user.**
- **Also mention that the user needs to run `yarn install`. There are dependencies being used in the app because it wouldn't run until I ran this command.**

### Happy coding 🥂.

**You reach me on slack (Rose Wanjohi).**

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hello Team :wave:
Status: Changes Required :recycle:

Good job :+1: your project looks nice, it works, no linter issue :smile: but before approving this project let's cover a few details together to make your project better :blush: :arrow_double_down: .

REQUIRED:

- [ ] Readme file: before `rails db:migrate` user must create the database using: rails `db:create`. Please add it.
- [ ] Readme file: To run the project in order to open it in the browser, user must use `rails server` command.
- [ ] Please remove all comments in your model and controller.
- [ ] As my teammate said: You can initialize user with just `@user = User.new`

OPTIONAL:

- [ ] Partial created for user form and reused in edit and new view
- [ ]  Validate the email format by using either a regular expression or a built in URI method.
- [ ] Form view to displays a list of the error messages (with #errors and full_messages usage) 

That's it, you can submit another code review request after making these changes
With all my respect :pray:

Rindra Josia

**Review by whiz25 — APPROVED** [review]
## Approved ✅ 
Well done for making the necessary changes as requested: Readme has been updated, comments removed from the codebase.
Merge this PR and proceed to the next milestone.


### Martin Afani

**Pull request merged**